### ISSUE TITLE
[ISSUE] Issue with Unity Catalog resources

### ISSUE THREAD

**Issue by zshao9** [databricks/terraform-provider-databricks#3013]
I need some help on understanding the following from https://registry.terraform.io/providers/databricks/databricks/latest/docs#special-configuration-for-unity-catalog:

```
Special configuration for Unity Catalog

Except for metastore, metastore assignment and storage credential objects, Unity Catalog APIs are accessible via workspace-level APIs. This design may change in the future.
```

Are we saying that:
* metastore, metastore assignment and storage credential objects are only accessible via account-level API;
All other Unity Catalog APIs are accessible via only workspace-level API?
* This technically doesn't make sense since Unity Catalog doesn't belong to a workspace (and multiple workspace can share the same unity catalog API).

**This is very cumbersome to use right now, because I will need to go back and forth between configuring workspace(s) and metastore.**
Can we make all Unity Catalog APIs accessible via account-level API?  We can keep the workspace-level API for backward-compatibility.

### Expected Behavior
All metastore resources should be able to be provisioned via account-level API.
We can keep workspace-level API for backward compatibility.

### Actual Behavior
Most of the metastore resources can only be provisioned via workspace-level API.

### Terraform and provider versions
1.31.1 (latest)

**Comment by zshao9** [comment]
@alexott What do you think?  I suspect that this might require a code copy from the workspace API server to the account API server, so there will be some work.  But it's the right thing to do, right?

**Comment by alexott** [comment]
This says:

* Metastore and storage credentials resources are available on both account and workspace levels ([src1](https://github.com/databricks/terraform-provider-databricks/blob/master/catalog/resource_metastore.go#L69), [src2](https://github.com/databricks/terraform-provider-databricks/blob/master/catalog/resource_storage_credential.go#L57)). But as I remember there were some nuances (@nkvuong knows this part better, and it's API-specific, not TF specific)
* Resources like catalogs, schemas, etc., are provisioned via workspace-level providers

It's all specific to the [available REST APIs](https://docs.databricks.com/api/), not the Terraform provider.

**Comment by zshao9** [comment]
Yeah I figured that the limitation is more on the REST API side.  Who shall
I reach out for REST API changes?

By the way, I believe there is still a bug in the Terraform provider: In
most cases, it simply *ignores the `metastore_id` attribute of a resource*
specification and default to the one assigned to the workspace.
IMO it should simply error out if the *metastore_id* doesn't match the
workspace's default metastore.

**Comment by zshao9** [comment]
See an example
<https://registry.terraform.io/providers/databricks/databricks/latest/docs/resources/catalog#example-usage>
where `metastore_id` is an attribute:

```
resource "databricks_catalog" "sandbox" {
  metastore_id = databricks_metastore.this.id
  name         = "sandbox"
  comment      = "this catalog is managed by terraform"
  properties = {
    purpose = "testing"
  }
}
```

**Comment by alexott** [comment]
For TF problems, file a separate GH issue (it could be API limitation as well). For API problems, reach solutions architect or professional services who work with you as I know...

**Comment by zshao9** [comment]
@alexott Submitted a separate ticket for TF: https://github.com/databricks/terraform-provider-databricks/issues/3014

I will follow up with SA on the API issue.

**Comment by KalleBylin** [comment]
Would like to +1 this issue. I have been implementing a new Unity Catalog and we want to manage certain resources across workspaces.

One example of this is using catalogs to provide segregation across the information architecture [as described here](https://github.com/databricks/terraform-provider-databricks/issues/3013). 

This gives the impression that catalogs can be managed independently of workspaces, which makes sense. It also makes sense to sometimes manage shared catalogs like bronze, silver, gold closer to the UC resources. But as it is now we are forced to have a dependency on a workspace and manage these resources closer to a workspace even if logically these resources don't "belong" to that workspace.
